### PR TITLE
Refactoring the code in Memorize for standard implementation of presence

### PR DESCRIPTION
**Pull request by utkarsh-raj**

### CODE REVIEW

**Comment by utkarsh-raj** [comment]
I supposed to keep both the commits so that it becomes easier track changes to individual files in the future. Possible fix to #374

File: activities/Memorize.activity/js/activity.js [hunk 1]
```diff
@@ -21,6 +21,107 @@ define(function (require) {
...
+            div.innerHTML = text;
+            document.body.appendChild(div);
+            tmpTimeout = setTimeout(function () {
+                tmpTimeout = null;
+                div.parentElement.removeChild(div);
+                tmpDiv = null;
+            }, 3000);
+        }
+
+        presencepalette.PresencePalette.prototype.onSharedActivityUserChanged = function (msg) {
```
Comment (llaske) [inline activities/Memorize.activity/js/activity.js]: I'm not a big fan of the the idea to update the PresencePalette prototype.
Change on users list should be detected with the event onSharedActivityUserChanged on the presence object, see for example https://github.com/llaske/sugarizer/blob/dev/activities/MazeWeb.activity/js/activity.js#L35

Comment (utkarsh-raj) [reply activities/Memorize.activity/js/activity.js]: I have identified the mistake ! Indeed, one mus t not make changes to the prototype during runtime. I have made the necessary changes by assign callback to the mentioned event.

**Comment by utkarsh-raj** [comment]
Made the requested changes. Possibly the callback can be used to respond to the event trigger.

**Comment by llaske** [comment]
I've got two errors on the console when I'm running the activity:

```
activity.js:25 Uncaught ReferenceError: env is not defined
    at activity.js:25
    at Object.execCb (require.js:1604)
    at Module.check (require.js:844)
    at Module.<anonymous> (require.js:1084)
    at require.js:129
    at require.js:1127
    at each (require.js:57)
    at Module.emit (require.js:1126)
    at Module.check (require.js:898)
    at Module.<anonymous> (require.js:1084)
(anonymous) @ activity.js:25
execCb @ require.js:1604
check @ require.js:844
(anonymous) @ require.js:1084
(anonymous) @ require.js:129
(anonymous) @ require.js:1127
each @ require.js:57
emit @ require.js:1126
check @ require.js:898
(anonymous) @ require.js:1084
(anonymous) @ require.js:129
(anonymous) @ require.js:1127
each @ require.js:57
emit @ require.js:1126
check @ require.js:898
enable @ require.js:1114
init @ require.js:757
callGetModule @ require.js:1141
completeLoad @ require.js:1498
onScriptLoad @ require.js:1625
load (async)
req.load @ require.js:1830
load @ require.js:1593
load @ require.js:803
fetch @ require.js:793
check @ require.js:823
enable @ require.js:1114
enable @ require.js:1465
(anonymous) @ require.js:1099
(anonymous) @ require.js:129
each @ require.js:57
enable @ require.js:1061
init @ require.js:757
callGetModule @ require.js:1141
completeLoad @ require.js:1498
onScriptLoad @ require.js:1625
load (async)
req.load @ require.js:1830
load @ require.js:1593
load @ require.js:803
fetch @ require.js:793
check @ require.js:823
enable @ require.js:1114
enable @ require.js:1465
(anonymous) @ require.js:1099
(anonymous) @ require.js:129
each @ require.js:57
enable @ require.js:1061
init @ require.js:757
(anonymous) @ require.js:1380
setTimeout (async)
req.nextTick @ require.js:1709
localRequire @ require.js:1369
requirejs @ require.js:1691
(anonymous) @ activity.js:3
execCb @ require.js:1604
check @ require.js:844
enable @ require.js:1114
init @ require.js:757
callGetModule @ require.js:1141
completeLoad @ require.js:1498
onScriptLoad @ require.js:1625
load (async)
req.load @ require.js:1830
load @ require.js:1593
load @ require.js:803
fetch @ require.js:793
check @ require.js:823
enable @ require.js:1114
enable @ require.js:1465
(anonymous) @ require.js:1099
(anonymous) @ require.js:129
each @ require.js:57
enable @ require.js:1061
init @ require.js:757
(anonymous) @ require.js:1380
setTimeout (async)
req.nextTick @ require.js:1709
localRequire @ require.js:1369
requirejs @ require.js:1691
(anonymous) @ loader.js:10
activity.js:181 Uncaught TypeError: presencePalette.onUsersListChanged is not a function
    at activity.js:181
    at activity.js:121
    at WebSocket.that.socket.onopen (presence.js:151)
(anonymous) @ activity.js:181
(anonymous) @ activity.js:121
that.socket.onopen @ presence.js:151
```

Plus the idea is to replace the `presencepalette.js` file by the standard one in `ActivityTemplate` directory.

**Comment by utkarsh-raj** [comment]
Made the necessary changes and tested on Local. The presencepalette.js is same in functionalities with the ActivityTemplate file.

**Comment by llaske** [comment]
There is no error in console but now the Presence palette is no longer clickable :-(
Did you test the activity in shared mode?
Plus, the presencepalette.js file is not exactly the same than the one in ActivityTemplate. To ensure that the implementation is standard, you should replace the presencepalette.js file in the activity by the one in ActivityTemplate.

**Comment by utkarsh-raj** [comment]
I have made some changes. The Activity should work as desired. The implementation in the presencePalette.js file has also been made same @llaske

**Comment by llaske** [comment]
The presence palette is there now.
By the way, when I'm sharing the activity, I've got a message on the console:

```
activity.js:100 Uncaught ReferenceError: generateXOLogoWithColor is not defined
    at Array.onSharedActivityUserChangedCallback (activity.js:100)
    at WebSocket.socket.onmessage (presence.js:56)
```

**Comment by utkarsh-raj** [comment]
Probably was a scope issue @llaske

**Comment by llaske** [comment]
I've now an error when I'm sharing the activity.

image

You must test the activity before sending your fix. It should work without error and without message in the console.

**Comment by utkarsh-raj** [comment]
The branch is changed. Closing.

**Pull request closed without merging**